### ISSUE TITLE
Named default import symbols get incorrectly renamed

### ISSUE THREAD

**Issue by seivan** [denoland/deno#9868]
There seems to be a distinction between the names of an exported value between  `deno run`  and `deno compile` when importing an object and using it as a key in a new object

In `'https://deno.land/x/yargs_parser@v20.2.4-deno/deno.ts'`
```ts
export default yargsParser
``` 

In `'https://deno.land/x/yargs/deno.ts'`
```ts
import Parser from 'https://deno.land/x/yargs_parser@v20.2.4-deno/deno.ts';

const newObject = {
 Parser
}
``` 
This will create `{yargParser: Parser}` in `deno compile` which is unexpected. 

Expected value would be `{Parser: Parser}` which is the result you get with `deno run`

Related: https://github.com/yargs/yargs/pull/1891

**Comment by kitsonk** [comment]
What problem is this causing?

When you bundle ESM modules, some renaming has to take place.  Modules are bundled before they can be used in `deno compile`.

**Comment by seivan** [comment]
@kitsonk Sorry for being unclear, I rephrased the issue. 
The expectation is that this should work  
```ts
const someObj = {
 Parser
 }
 
 assert(someObj.Parser)
 ```
 both in `deno run` and `deno compile`

**Comment by kitsonk** [comment]
cc/ @kdy1

Looks like some default import symbols get renamed, which then causes unexpected/breaking changes in the downstream code.

**Comment by nayeemrmn** [comment]
Occurs with any aliased imports:
```ts
import { foo } from './temp2.ts';

console.log({
  foo
});
```
```ts
// temp2.ts
const bar = 1;
export { bar as foo };
```
Bundle output:
```js
console.log({
    bar: 1
});
```
Expected:
```js
console.log({
    foo: 1
});
```

SWC should ensure `{ foo }` is treated the same as `{ foo: foo }`.

**Comment by kdy1** [comment]
Fixed on upstream. (Published a new version)

**Comment by bartlomieju** [comment]
Fixed in 1.9.0
